src/permutat.c

```diff
@@ -4454,23 +4455,47 @@ Obj FuncMappingPermListList(Obj self, Obj src, Obj dst)
     }
     d = 0;
     for (i = 1;i <= l;i++) {
-        x = INT_INTOBJ(ELM_LIST(src,i));
+        obj = ELM_LIST(src, i);
+        if (!IS_POS_INTOBJ(obj)) {
+            ErrorMayQuit("Lists can only contain positive integers", 0L, 0L);
```

Comment:
This error message is a bit too general, how about "The first argument must be a list of positive integers", arguably lists in GAP can contain things other than positive integers, just not in this function.